### ISSUE TITLE
Declarator blocks not working for unit-declared roles.

### ISSUE THREAD

**Issue by JJ** [rakudo/rakudo#3521]
## The Problem

If you add an declarator block to a Role: 

```perl6
#| Why
unit role NoWhy;
```
It does not seem to work:

```
No documentation available for type 'NoWhy'.
Perhaps it can be found at https://docs.raku.org/type/NoWhy
```

## Expected Behavior

As in the case of classes or modules, it should work.

## Environment
* Operating system: 
* Compiler version (`perl6 -v`): 

Ubuntu 16.04, tested with 2020.02 and 2020.01 and down to 2019.03.1

**Comment by raiph** [comment]
Fwiw, in a 2018.04.1 rakudo it doesn't work for *any* form of `role`: https://glot.io/snippets/fl3ri2wfrc

**Comment by jnthn** [comment]
The reason is that the name installed for a role is actually a role *group*, not the role itself. That lives within the group. Often, however, we try to blur the difference a bit, so the role group behaves as if it was the unsignatured role in the group. We do that for quite a few things, but not `.WHY`. Making us delegate `.WHY` that way is easy enough, and I've implemented it. However it turns out there are three spectests that explicitly expect that we do *not* do that.

I'll make a PR with the changes I did so we can discuss a bit if there's a good reason to not do this change.
